dspace-api/src/main/java/org/dspace/scripts/filepreview/FilePreview.java

```diff
@@ -0,0 +1,165 @@
+            // Generate the file previews for the specified item with the given UUID.
+            handler.logInfo("\nGenerate the file previews for the specified item with the given UUID: " +
+                    specificItemUUID);
+        }
+    }
+
+    @Override
+    public void internalRun() throws Exception {
+        if (info) {
+            printHelp();
+            return;
+        }
+
+        Context context = new Context();
+        context.turnOffAuthorisationSystem();
+        if (StringUtils.isNotBlank(specificItemUUID)) {
+            // Generate the preview only for a specific item
+            generateItemFilePreviews(context, UUID.fromString(specificItemUUID));
+        } else {
+            // Generate the preview for all items
+            Iterator<Item> items = itemService.findAll(context);
+
+            int count = 0;
+            while (items.hasNext()) {
+                count++;
+                Item item = items.next();
+                try {
+                    generateItemFilePreviews(context, item.getID());
+                } catch (SQLException | AuthorizeException | IOException | ParserConfigurationException |
+                        ArchiveException | SAXException e) {
+                    handler.logError("Error while generating preview for item with UUID: " + item.getID());
+                    handler.logError(e.getMessage());
+                }
+
+                if (count % 100 == 0) {
+                    handler.logInfo("Processed " + count + " items.");
+                }
+            }
+        }
+        context.restoreAuthSystemState();
+        context.commit();
+        context.complete();
+    }
+
+    private void generateItemFilePreviews(Context context, UUID itemUUID) throws SQLException, AuthorizeException,
+            IOException, ParserConfigurationException, ArchiveException, SAXException {
+        Item item = itemService.find(context, itemUUID);
+        if (Objects.isNull(item)) {
+            handler.logError("Item with UUID: " + itemUUID + " not found.");
+            return;
+        }
+
+        List<Bundle> bundles = item.getBundles();
+        for (Bundle bundle : bundles) {
+            List<Bitstream> bitstreams = bundle.getBitstreams();
+            for (Bitstream bitstream : bitstreams) {
+                boolean canPreview = previewContentService.findOutCanPreview(context, bitstream);
+                if (!canPreview) {
+                    return;
+                }
+                List<FileInfo> fileInfos = new ArrayList<>();
+                List<PreviewContent> prContents = previewContentService.findRootByBitstream(context,
+                        bitstream.getID());
+                // Generate new content if we didn't find any
+                if (!prContents.isEmpty()) {
+                    return;
+                }
+
+                fileInfos = previewContentService.getFilePreviewContent(context, bitstream, fileInfos);
```

Comment:
why 2x fileinfos, ?

Reply:
Yes, because during creating preview there is created a new ArrayList https://github.com/dataquest-dev/DSpace/blob/dtq-dev/dspace-api/src/main/java/org/dspace/util/FileTreeViewGenerator.java#L70